portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +25,53 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  
+  /**
+   *This welcome String will remain here for now as it is still 
+   *in one of my PRs and I am unsure as to what the best practice 
+   *is for these cases. In the meantime, this String will be sent as
+   *part of the JSON response.
+  */
+  private String welcome = "Welcome to my portfolio!";
+  private ArrayList<String[]> data;
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    //Initialize Linked List/s
+    data = new ArrayList<String[]>();
+    buildIntroData();
+    
+    //Convert the List/s to JSON
+    String json = convertToJson(data);
+
+    //Send JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  //Builds the intro-data Linked List.
+  /**
+   * I implemented the data this way in order to make it easier for me to extend this implementation to the rest of
+   * the portfolio's pages. This way, I can just have my data contain various arrays for the different pages/data I
+   * need to load onto the page.
+   */
+  private void buildIntroData() {
+    String[] intro_page = new String[4];
```

Comment:
You could also define a Java object and then convert that to JSON. For example, you might have:
```
public class IntroPage {
  String imageSource;
  String message;
...
}
```